crates/sui-config/src/dynamic_transaction_signing_checks.rs

```diff
@@ -0,0 +1,262 @@
+    /// the transaction should be denied. Any other return value other than `True` (including
+    /// errors) should be considered a denial.
+    pub fn new(starlark_program: String) -> Result<Self, DynamicCheckRunnerError> {
+        // Adds global functions and variables to the dialect (e.g., True, False, Maps, Lists, etc.)
+        // The full spec of what exactly is added here can be found here:
+        // https://github.com/bazelbuild/starlark/blob/master/spec.md#built-in-constants-and-functions
+        let globals = Globals::standard();
+        warn!(
+            "Dynamic transaction checks are enabled. Make sure that you intend to be running \
+            dynamic checks on transactions."
+        );
+        let module = AstModule::parse(STAR_INPUT_FILE_NAME, starlark_program.clone(), &DIALECT)
+            .map_err(|e| DynamicCheckRunnerError::LoadingError(e.into_kind()))?;
+        Ok(Self {
+            module,
+            globals,
+            loaded_program: starlark_program,
+        })
+    }
+
+    /// Run the Starlark program in `self` with the given transaction data, signatures, input
+    /// object kinds, and receiving objects.
+    pub fn run_predicate(
+        &self,
+        tx_data: &TransactionData,
+        tx_signatures: &[GenericSignature],
+        input_object_kinds: &[InputObjectKind],
+        receiving_objects: &[ObjectRef],
+    ) -> Result<(), DynamicCheckRunnerError> {
+        let tx_data_json = serde_json::to_value(tx_data)
+            .map_err(|e| DynamicCheckRunnerError::JSONSerializationError(e.to_string()))?;
+        let tx_signatures_json = serde_json::to_value(tx_signatures)
+            .map_err(|e| DynamicCheckRunnerError::JSONSerializationError(e.to_string()))?;
+        let input_object_kinds_json = serde_json::to_value(input_object_kinds)
+            .map_err(|e| DynamicCheckRunnerError::JSONSerializationError(e.to_string()))?;
+        let receiving_objects_json = serde_json::to_value(receiving_objects)
+            .map_err(|e| DynamicCheckRunnerError::JSONSerializationError(e.to_string()))?;
+        self.run_starlark_predicate(
+            &tx_data_json,
+            &tx_signatures_json,
+            &input_object_kinds_json,
+            &receiving_objects_json,
+        )
+    }
+
+    fn run_starlark_predicate(
+        &self,
+        tx_data: &JsonValue,
+        tx_signatures: &JsonValue,
+        tx_input_object_kinds: &JsonValue,
+        tx_receiving_objects: &JsonValue,
+    ) -> Result<(), DynamicCheckRunnerError> {
+        let heap = Heap::new();
+        let env = Module::new();
+
+        let tx_data_value = Self::json_to_starlark(tx_data, &heap)?;
+        let tx_signers_value = Self::json_to_starlark(tx_signatures, &heap)?;
+        let tx_input_object_kinds_value = Self::json_to_starlark(tx_input_object_kinds, &heap)?;
+        let tx_receiving_objects_value = Self::json_to_starlark(tx_receiving_objects, &heap)?;
+
+        env.set(TX_DATA_NAME, tx_data_value);
```

Comment:
Can we also provide the transaction digest?

Reply:
We can! Added :)